algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/workload.py

```diff
@@ -0,0 +1,253 @@
+        augmented_and_preprocessed_input_batch['input_paddings'])
+
+    return (logits, logits_paddings), None
+
+  @property
+  def model_params_types(self):
+    if self._param_types is None:
+      self._param_types = param_utils.pytorch_param_types(self._param_shapes)
+    return self._param_types
+
+  def build_input_queue(self,
+                        data_rng: jax.random.PRNGKey,
+                        split: str,
+                        data_dir: str,
+                        global_batch_size: int,
+                        num_batches: Optional[int] = None,
+                        repeat_final_dataset: bool = False):
+    per_device_batch_size = int(global_batch_size / N_GPUS)
+    n_inputs = 4
+    keys = ['inputs', 'input_paddings', 'targets', 'target_paddings']
+    np_iter = super().build_input_queue(data_rng,
+                                        split,
+                                        data_dir,
+                                        global_batch_size,
+                                        num_batches,
+                                        repeat_final_dataset)
+    while True:
+      # Only iterate over tf input pipeline in one Python process to
+      # avoid creating too many threads.
+      if RANK == 0:
+        batch = next(np_iter)  # pylint: disable=stop-iteration-return
+        tensor_list = []
+        for key in keys:
+          value = batch[key]
+          tensor = torch.as_tensor(value, dtype=torch.float32, device=DEVICE)
+          tensor_list.append(tensor)
+          batch[key] = (
+              tensor[0] if USE_PYTORCH_DDP else tensor.view(
+                  -1, *value.shape[2:]))
+        # Send batch to other devices when using DDP.
+        if USE_PYTORCH_DDP:
+          # During eval, the batch size of the remainder might be different.
+          if split != 'train':
+            per_device_batch_size = torch.tensor(
+                len(batch['inputs']), dtype=torch.int32, device=DEVICE)
+            dist.broadcast(per_device_batch_size, src=0)
+          dist.broadcast(torch.cat(tensor_list, dim=-1), src=0)
+      else:
+        # During eval, the batch size of the remainder might be different.
+        if split != 'train':
+          per_device_batch_size = torch.empty((1,),
+                                              dtype=torch.int32,
+                                              device=DEVICE)
+          dist.broadcast(per_device_batch_size, src=0)
+        tensor = torch.empty(
+            (N_GPUS, per_device_batch_size, 320000 * 2 + 256 * 2),
+            dtype=torch.float32,
+            device=DEVICE)
+        dist.broadcast(tensor, src=0)
+        # Note that the order of the keys is important.
+        tensors = tensor.split([320000, 320000, 256, 256], dim=-1)
+        batch = {}
+        for key, n in zip(keys, range(n_inputs)):
+          batch[key] = tensors[n][RANK]
+      yield batch
+
+  def compute_loss(self, logits, logit_paddings, targets, target_paddings):
```

Comment:
is this supposed to be `loss_fn` ?

Reply:
In order to resolve style errors related to changing argument names in subclass implementation, I renamed `loss_fn` to `compute_loss` for this workload and totally forgot to add a dummy method to override the abstract method `loss_fn` of the super class: what do you suggest we do?

Reply:
you can leave it as is for now and we can refactor it later

(I had refactored the data iterator for librispeech to produce tuple(input, input_padding) and tuple(target, target_padding), and then loss_fn can take in logits_batch=tuple(logit, logit_padding) and targets_batch=(targets, target_padding), but we can fix that once this is checked in)